src/pluto/grey_image.cr

```diff
@@ -0,0 +1,63 @@
+require "./image"
+
+class Pluto::GreyImage < Pluto::Image
+  property grey : Array(UInt8)
+  property width : Int32
+  property height : Int32
+
+  def self.new(red : Array(UInt8), green : Array(UInt8), blue : Array(UInt8), alpha : Array(UInt8), width : Int32, height : Int32)
+    RGBImage.new(red, green, blue, alpha, width, height).to_grey
+  end
+
+  def initialize(@grey, @width, @height)
+  end
+
+  def clone : GreyImage
+    self.class.new(
+      @grey.clone,
+      @width,
+      @height
+    )
+  end
+
+  def red : Array(UInt8)
+    @grey
+  end
+
+  def green : Array(UInt8)
+    @grey
+  end
+
+  def blue : Array(UInt8)
+    @grey
+  end
+
+  def alpha : Array(UInt8)
+    @grey
+  end
+
+  def each_channel(& : (Array(UInt8), ChannelType) -> Nil) : Nil
+    yield @grey, ChannelType::Grey
+    nil
+  end
+
+  def [](channel_type : ChannelType) : Array(UInt8)
+    # All channels are grey. There can only be grey.
+    @grey
```

Comment:
Can we raise here as we do in `#[]=`?

Reply:
I initially didn't do that so that this method would behave the same way as the `red`, `green`, and `blue` methods using the `[ChannelType::Red]`, `[ChannelType::Green]`, and `[ChannelType::Blue]` respectively. This was before the common parent class that helped enforce the existence of those previous methods, so it's probably not necessary anymore. Will add in a check :+1: